Wait until Adapter is available

eslesar opened the pull request:
This fix is taken directly from @PlagueHO fix at https://github.com/PlagueHO/LoopbackAdapter/pull/2/files

When xIPAddress follows xVMSwitch without any wait time, it causes an error:

PowerShell DSC resource MSFT_xIPAddress  failed to execute Test-TargetResource functionality with error message: The running command stopped because the preference variable 
"ErrorActionPreference" or common parameter is set to Stop: No matching MSFT_NetIPAddress objects found by CIM query for instances of the ROOT/StandardCimv2/MSFT_NetIPAddress class on the  
CIM server: SELECT * FROM MSFT_NetIPAddress  WHERE ((InterfaceAlias LIKE 'vEthernet (Internal)')) AND ((AddressFamily = 2)). Verify query parameters and retry.

In searching for a fix, I found the above-referenced issue in a  @PlagueHO repo. Following his example, this addition of a loop/test/wait fixes the problem.

msftclas commented:
Hi __@eslesar__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <p>
        It looks like you're a Microsoft contributor (Eric Slesar). If you're full-time, we DON'T require a Contribution License Agreement. If you are a vendor, please DO sign the electronic Contribution License Agreement. It will take 2 minutes and there's no faxing! https://cla.microsoft.com.
    </p>

TTYL, MSBOT;

PlagueHO commented:
Good catch @eslesar! That would actually explain some random problems I have building out my own labs using this resource. I never thought to try moving this code into xIPAddress -good thinking!

Can you add a change entry in the #Unreleased section of the Readme.md file as well for this please?

Thank you for working on this.

---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[DSCResources/MSFT_xIPAddress/MSFT_xIPAddress.psm1, line 198 at r1](https://reviewable.io:443/reviews/powershell/xnetworking/143#-KRpQnQL4JoGjoX0FQ7T:-KRpQnQL4JoGjoX0FQ7U:b-c9bleb) ([raw file](https://github.com/powershell/xnetworking/blob/c48639fb0f3141d6340e7d69758288c35ea455b6/DSCResources/MSFT_xIPAddress/MSFT_xIPAddress.psm1#L198)):*
> ```PowerShell
>     [DateTime] $StartTime = Get-Date 
> ```

Can you remove this blank line please? :smile: https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md#no-more-than-two-consecutive-newlines

---

*[DSCResources/MSFT_xIPAddress/MSFT_xIPAddress.psm1, line 214 at r1](https://reviewable.io:443/reviews/powershell/xnetworking/143#-KRpRXQDb4UUKNYXfjTC:-KRpRXQDb4UUKNYXfjTD:b-1h7gv9) ([raw file](https://github.com/powershell/xnetworking/blob/c48639fb0f3141d6340e7d69758288c35ea455b6/DSCResources/MSFT_xIPAddress/MSFT_xIPAddress.psm1#L214)):*
> ```PowerShell
>         }
>         catch
>         {
> ```

Technically I shouldn't have done this (https://github.com/PowerShell/DscResources/blob/master/BestPractices.md#avoid-empty-catch-blocks). But given the purpose of this function (to wait till no errors occur pulling the IP address) it is probably OK.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xnetworking/143#-:-KRpQb6R0JyNBASjqtxg:b-oyw4y2)*
<!-- Sent from Reviewable.io -->

eslesar commented:
I added a bullet to Readme.md, removed the try-catch (tested with -ErrorAction SilentlyContinue  instead and it works fine), and removed the extra line.

PlagueHO commented:
Good stuff - much cleaner! Thank you (I should probably copy this into LoopbackAdapter too - I'll add it to my backlog :smile: ).

@TravisEz13 or @TysonJHayes - care to review and merge when you can?

<img class="emoji" title=":LGTM:" alt=":LGTM:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 2 files at r2.
Review status: 1 of 2 files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xnetworking/143#-:-KRpgpVa1N6oSA5eZk0e:bpybxq2)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
Reviewed 1 of 2 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xnetworking/143)*
<!-- Sent from Reviewable.io -->

kwirkykat commented:
Reviewed 1 of 2 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[DSCResources/MSFT_xIPAddress/MSFT_xIPAddress.psm1, line 196 at r2](https://reviewable.io:443/reviews/powershell/xnetworking/143#-KS3q9IiEJNyjq_cfw45:-KS3q9IiEJNyjq_cfw46:bsc8n7y) ([raw file](https://github.com/powershell/xnetworking/blob/71f8d758215657e42187906841ecdc62199c03ae/DSCResources/MSFT_xIPAddress/MSFT_xIPAddress.psm1#L196)):*
> [Boolean] $AdapterBindingReady = $false
> [DateTime] $StartTime = Get-Date

These variables should not be capitalized since they are not parameters.
See [DSC Resource style guidelines](https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md#variable-names-use-camel-case).

---

*[DSCResources/MSFT_xIPAddress/MSFT_xIPAddress.psm1, line 208 at r2](https://reviewable.io:443/reviews/powershell/xnetworking/143#-KS3qbNA8LwCiSn82Cf4:-KS3qbNBYbXZOlPgGusJ:bhquoh3) ([raw file](https://github.com/powershell/xnetworking/blob/71f8d758215657e42187906841ecdc62199c03ae/DSCResources/MSFT_xIPAddress/MSFT_xIPAddress.psm1#L208)):*
>          Start-Sleep -Milliseconds 200

Can you put this code in an else so that it doesn't sleep if the binding is ready?
Like this:
```powershell
if ($currentIPs)
{
    $AdapterBindingReady = $true
}
else
{
    Start-Sleep -Milliseconds 200
}
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xnetworking/143)*
<!-- Sent from Reviewable.io -->

kwirkykat commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xnetworking/143#-:-KS3wQxCIIT_SDuADEnz:bnfp4nl)*
<!-- Sent from Reviewable.io -->
